site/src/sphinx/client-service-discovery.rst

```diff
@@ -0,0 +1,237 @@
+.. _client-service-discovery:
+
+Client-side load balancing and service discovery
+================================================
+
+You can configure an Armeria client to distribute its requests to more than one server autonomously, unlike
+traditional server-side load balancing where the requests goes through a dedicated load balancer such as
+`L4 and L7 switches <https://en.wikipedia.org/wiki/Multilayer_switch#Layer_4%E2%80%937_switch,_web_switch,_or_content_switch>`_.
+
+There are 4 elements involved in client-side load balancing in Armeria:
+
+- :api:`Endpoint` represents an individual host (with an optional port number) and its weight.
+- :api:`EndpointGroup` represents a set of :apiplural:`Endpoint`.
+- :api:`EndpointGroupRegistry` is a global registry of :apiplural:`EndpointGroup` where each
+  :api:`EndpointGroup` is identified by its unique name.
+- A user specifies the target group name in a URI,
+  e.g. ``http://group:my_group/`` where ``my_group`` is the group name.
```

Comment:
how about 

```
e.g.  ``http://group:group_name``  (examples: ``http://group:google/`` or ``http://group:facebook/``)
```

Reply:
`e.g.` means 'for example', so it seems weird to have an example in an example.

Reply:
Added a little bit more information for clarity